File: src/main/java/mods/eln/item/electricalitem/ElectricalPickaxe.java
package mods.eln.item.electricalitem;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;
import net.minecraft.item.ItemStack;

public class ElectricalPickaxe extends ElectricalTool {

    public ElectricalPickaxe(String name, float strengthOn, float strengthOff,
                             double energyStorage, double energyPerBlock, double chargePower) {
        super(name, strengthOn, strengthOff, energyStorage, energyPerBlock, chargePower);
    }

    @Override
    public float getStrVsBlock(ItemStack stack, Block block) {
        float value = block != null && (block.getMaterial() == Material.iron || block.getMaterial() == Material.glass || block.getMaterial() == Material.anvil || block.getMaterial() == Material.rock) ? getStrength(stack) : super.getStrVsBlock(stack, block);
        for (Block b : blocksEffectiveAgainst) {
            if (b == block) {
                value = getStrength(stack);
                break;
            }
        }
        // Utils.println("****" + value);
        return value;
    }
}

File: src/main/java/mods/eln/item/electricalitem/ElectricalPickaxe.kt
package mods.eln.item.electricalitem

import mods.eln.wiki.Data
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.item.Item
import net.minecraft.item.ItemStack

class ElectricalPickaxe(name: String, strengthOn: Float, strengthOff: Float,
                        energyStorage: Double, energyPerBlock: Double, chargePower: Double) : ElectricalTool(name, strengthOn, strengthOff, energyStorage, energyPerBlock, chargePower) {

    override fun setParent(item: Item?, damage: Int) {
        super.setParent(item, damage)
        Data.addPortable(newItemStack())
    }

    override fun getStrVsBlock(stack: ItemStack, block: Block?): Float {
        var value = if (block != null && (block.material === Material.iron || block.material === Material.glass || block.material === Material.anvil || block.material === Material.rock)) getStrength(stack) else super.getStrVsBlock(stack, block)
        for (b in blocksEffectiveAgainst) {
            if (b === block) {
                value = getStrength(stack)
                break
            }
        }
        return value
    }
}
